WooCommerce/Classes/ViewRelated/Orders/Payment Methods/PaymentMethodsViewModel.swift

```diff
@@ -268,7 +268,13 @@ final class PaymentMethodsViewModel: ObservableObject {
                 // Update order in case its status and/or other details are updated after a failed in-person payment
                 self?.updateOrderAsynchronously()
 
-                onFailure()
+                switch error {
+                case let error as CardPaymentErrorProtocol where error.requiresFallbackPaymentMethod:
```

Comment:
Just wondering: Should we put this switch-case filtering into a separate function? 🤔 Something like `shouldFallbackToOtherPaymentMethods(for discoveryMethod: CardReaderDiscoveryMethod, error: Error) -> Bool`

Reply:
Good idea 👍